src/transformers/modeling_tf_mpnet.py

```diff
@@ -0,0 +1,1111 @@
+        self.out = TFMPNetOutput(config, name="output")
+
+    def call(self, hidden_states, attention_mask, head_mask, output_attentions, position_bias=None, training=False):
+        self_attention_outputs = self.attention(
+            hidden_states, attention_mask, head_mask, output_attentions, position_bias=position_bias, training=training
+        )
+        attention_output = self_attention_outputs[0]
+        outputs = self_attention_outputs[1:]  # add self attentions if we output attention weights
+
+        intermediate_output = self.intermediate(attention_output)
+        layer_output = self.out(intermediate_output, attention_output, training=training)
+        outputs = (layer_output,) + outputs  # add attentions if we output them
+
+        return outputs
+
+
+class TFMPNetEncoder(tf.keras.layers.Layer):
+
+    def __init__(self, config, **kwargs):
+        super().__init__(**kwargs)
+
+        self.config = config
+        self.n_heads = config.num_attention_heads
+        self.output_attentions = config.output_attentions
+        self.output_hidden_states = config.output_hidden_states
+
+        self.layer = [TFMPNetLayer(config, name="layer_._{}".format(i)) for i in range(config.num_hidden_layers)]
+        self.relative_attention_bias = tf.keras.layers.Embedding(
+            config.relative_attention_num_buckets,
+            self.n_heads,
+            name="relative_attention_bias",
+        )
+        self.relative_attention_num_buckets = config.relative_attention_num_buckets
+    
+    def call(
+        self,
+        hidden_states,
+        attention_mask,
+        head_mask,
+        output_attentions,
+        output_hidden_states,
+        return_dict,
+        training=False,
+    ):
+        position_bias = self.compute_position_bias(hidden_states)
+        all_hidden_states = () if output_hidden_states else None
+        all_attentions = () if output_attentions else None
+
+        for i, layer_module in enumerate(self.layer):
+            if output_hidden_states:
+                all_hidden_states = all_hidden_states + (hidden_states,)
+
+            layer_outputs = layer_module(
+                hidden_states, attention_mask, head_mask[i], output_attentions, position_bias=position_bias, training=training
+            )
+            hidden_states = layer_outputs[0]
+
+            if output_attentions:
+                all_attentions = all_attentions + (layer_outputs[1],)
+
+        # Add last layer
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        if not return_dict:
+            return tuple(v for v in [hidden_states, all_hidden_states, all_attentions] if v is not None)
+
+        return TFBaseModelOutput(
+            last_hidden_state=hidden_states, hidden_states=all_hidden_states, attentions=all_attentions
+        )
+
+    # Copied from transformers.modeling_tf_t5.TFT5Attention
```

Comment:
This mention needs the name of the method (those are used by our internal scripts to check the conformance of the copy to the original).